ocs_ci/utility/multicluster.py

```diff
@@ -237,11 +196,71 @@ def generate_pytest_parameters(self, metafunc, roles):
         """
         pytest_params = []
         for role in roles:
-            pytest_params.extend(self.get_pytest_params_tuple(role))
+            # A role marker may have mdr, rdr etc markers, we need to pick
+            # the role marker only if its applicable to the current DR scenario
+            if role.startswith(f"{self.dr_type}"):
+                pytest_params.extend(self.get_pytest_params_tuple(role))
         return pytest_params
 
 
-multicluster_upgrade_parametrizer = {"metro-dr": MDRClusterUpgradeParametrize}
+class MDRClusterUpgradeParametrize(MultiClusterUpgradeParametrize):
+    """
+    This child class handles MDR upgrade scenario specific pytest parametrization
+
+    """
+
+    def __init__(self):
+        self.dr_type = "mdr"
+        super().__init__()
+        self.all_roles = MDR_ROLES
+
+    def config_init(self):
+        super().config_init()
+
+    def generate_role_ranks(self):
+        """
+        Based on current roles for MDR : ActiveACM:1, PassiveACM:1, PrimaryODF:2, SecondaryODF: 2
+
+        """
+        # For now we will stick to this convention
+        self.role_ranks = {
+            "ActiveACM": 1,
+            "PassiveACM": 1,
+            "PrimaryODF": 2,
+            "SecondaryODF": 2,
+        }
+
+
+class RDRClusterUpgradeParametrize(MultiClusterUpgradeParametrize):
+    """
+    This child class handles RDR upgrade scenario specific pytest parametrization
+
+    """
+
+    def __init__(self):
+        self.dr_type = "rdr"
+        super().__init__()
+        self.all_roles = RDR_ROLES
+
+    def config_init(self):
+        super().config_init()
+
+    def generate_role_ranks(self):
+        """
+        Based on the current roles for RDR : ActiveACM:1, PrimaryODF:2, SecondaryODF: 2
+
+        """
+        self.role_ranks = {
+            "ActiveACM": 1,
```

Comment:
As we discussed, we can also add directly the PassiveACM role rank here as well.

Reply:
done